api/app/notifications/notifications_ses_callback.py

```diff
@@ -12,11 +17,78 @@
 )
 from app.dao.service_callback_api_dao import get_service_callback_api_for_service
 from app.notifications.process_client_response import validate_callback_data
+from app.notifications.utils import autoconfirm_subscription
 from app.celery.service_callback_tasks import (
     send_delivery_status_to_service,
     create_encrypted_callback_data,
 )
 from app.config import QueueNames
+from app.errors import (
+    register_errors,
+    InvalidRequest
+)
+import validatesns
+
+ses_callback_blueprint = Blueprint('notifications_ses_callback', __name__)
+
+register_errors(ses_callback_blueprint)
+
+
+class SNSMessageType(enum.Enum):
+    SubscriptionConfirmation = 'SubscriptionConfirmation'
+    Notification = 'Notification'
+    UnsubscribeConfirmation = 'UnsubscribeConfirmation'
+
+
+class InvalidMessageTypeException(Exception):
+    pass
+
+
+def verify_message_type(message_type: str):
+    try:
+        SNSMessageType(message_type)
+    except ValueError:
+        raise InvalidMessageTypeException(f'{message_type} is not a valid message type.')
+
+
+certificate_cache = dict()
```

Comment:
we should use one of the flask cache systems:

https://werkzeug.palletsprojects.com/en/0.15.x/contrib/cache/#cache-systems

`simplecache` is not great but better than a dict; and can be migrated to a `rediscache` when we're able to support it

Reply:
http://flask.pocoo.org/docs/1.0/patterns/caching/

Reply:
@dagoof thanks for that—I'll pull in a simple cache for now.